tests/functional/common.js

```diff
@@ -103,6 +103,18 @@ export function execWebExt(
   spawnOptions: child_process$spawnOpts,
 ): RunningWebExt {
 
+  if (spawnOptions.env) {
+    spawnOptions.env = {
+      // Propagate the current environment when redifining it from the spawnOptions
+      // otherwise it may trigger unexpected failures due to missing variables that
+      // are expected to be part of the environment (e.g. #2444 was failing only on
+      // windows because of this, due @pnpm/npm-conf, a transitive dependencies for
+      // update-notifier, expecting process.env.APPDATA to be defined when running
+      // on windows).
```

Comment:
```suggestion
      // Propagate the current environment when redefining it from the `spawnOptions`
      // otherwise it may trigger unexpected failures due to missing variables that
      // are may be expected (e.g. #2444 was failing only on Windows because
      // @pnpm/npm-conf, a transitive dependencies for update-notifier, was expecting
      // process.env.APPDATA to be defined when running on Windows).
```